### PR TITLE
Sync Blood Pressures

### PR DESCRIPTION
**Pull request by ssrihari**
Issue: https://github.com/resolvetosavelives/redapp/issues/17

### CODE REVIEW

**Comment by ssrihari** [comment]
All the duplication in the code until now is present intentionally. Will refactor.

File: app/models/blood_pressure.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class BloodPressure < ApplicationRecord
+  include Mergeable
+  belongs_to :patient, optional: true
+
+  def with_payload_keys(attributes)
```
Comment (timcheadle) [inline app/models/blood_pressure.rb]: Seems like we could turn this into a model concern if we're copying this code into multiple models.

File: spec/controllers/api/v1/blood_pressures_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  describe 'POST sync: send data from device to server;' do
+
+    describe 'creates new blood pressures' do
+
+      it 'returns 400 when there are no blood pressures in the request' do
+        post(:sync_from_user, params: { blood_pressures: [] })
+        expect(response.status).to eq 400
+      end
+
+      it 'creates new blood pressures without associated patient' do
```
Comment (timcheadle) [inline spec/controllers/api/v1/blood_pressures_controller_spec.rb]: Hmm... is this something we want? Creating BPs without a patient?

File: spec/requests/blood_pressure_sync_spec.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    get sync_route, params: {}, headers: headers
+
+    response_body = JSON(response.body)
+    expect(response.status).to eq 200
+    expect(response_body['blood_pressures']).to eq([])
+    expect(response_body['processed_since']).to eq(Time.new(0).strftime('%FT%T.%3NZ'))
+  end
+
+  it 'push a new valid blood_pressure and pull first time' do
```
Comment (timcheadle) [inline spec/requests/blood_pressure_sync_spec.rb]: Typo: should be "pushes"

File: spec/controllers/api/v1/blood_pressures_controller_spec.rb (hunk 2 above)
Comment (govindkrjoshi) [reply spec/controllers/api/v1/blood_pressures_controller_spec.rb]: This is intentional. 
BPs will have weak references to patients so can independently sync patients and blood pressures.

**Pull request merged**